core/src/main/java/org/sonarsource/sonarlint/core/websocket/SonarCloudWebSocket.java

```diff
@@ -46,12 +47,17 @@ public class SonarCloudWebSocket {
   private static final Gson gson = new Gson();
   private WebSocket ws;
   private final History history = new History();
-  private final ScheduledExecutorService messageHistoryCleaner = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "sonarcloud-websocket-history-cleaner"));
+  private final ScheduledExecutorService sonarCloudWebSocketScheduler = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "sonarcloud-websocket-scheduled-jobs"));
+  private static ScheduledFuture<?> scheduledConnectionRefresh;
 
-  public static SonarCloudWebSocket create(HttpClient httpClient, Consumer<ServerEvent> serverEventConsumer) {
+  public static SonarCloudWebSocket create(HttpClient httpClient, Consumer<ServerEvent> serverEventConsumer, Runnable connectionRefresher) {
     var webSocket = new SonarCloudWebSocket();
     webSocket.ws = httpClient.createWebSocketConnection(WEBSOCKET_DEV_URL, rawEvent -> webSocket.handleRawMessage(rawEvent, serverEventConsumer));
-    webSocket.messageHistoryCleaner.scheduleAtFixedRate(webSocket::cleanUpMessageHistory, 0, 1, TimeUnit.MINUTES);
+    webSocket.sonarCloudWebSocketScheduler.scheduleAtFixedRate(webSocket::cleanUpMessageHistory, 0, 1, TimeUnit.MINUTES);
+    if(scheduledConnectionRefresh != null) {
+      scheduledConnectionRefresh.cancel(true);
+    }
+    scheduledConnectionRefresh = webSocket.sonarCloudWebSocketScheduler.scheduleAtFixedRate(connectionRefresher, 119, 119, TimeUnit.MINUTES);
```

Comment:
What about scheduling it for a single run instead of fixed rate? Anyway when the task is run, the socket will be closed and the executor shutdown. This way there is no need to cancel the recurring task

Reply:
Yes, I can schedule it for one time in 119 minutes, but I guess cancellation is still needed? In case connection creation is called during the 119 minutes, we would reschedule to start 119 minutes after the newly created time